FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/syscall.c

```diff
@@ -61,6 +59,34 @@ void uart_init()
     UART0_ADDR->CTRL = UART_CTRL_TX_EN;
 }
 
+#ifdef __PICOLIBC__
+
+#include <stdio.h>
+
+/**
+ * @brief  Write byte to the UART channel to be displayed on the command line
+ *         with qemu
+ * @param [in] c     byte to send
+ * @param [in] file  ignored
+ * @returns the character written (cast to unsigned so it is not an error value)
+ */
+
+int
+_uart_putc(char c, FILE *file)
+{
+    (void) file;
+    UART_DR( UART0_ADDR ) = c;
+    return (unsigned char) c;
+}
+
+static FILE __stdio = FDEV_SETUP_STREAM(_uart_putc, NULL, NULL, _FDEV_SETUP_WRITE);
```

Comment:
Note: We should probably switch these demos over to using semihosting properly for stdout. I'll give this a go in another PR.

Reply:
I figured it was simplest to leave the sample mostly alone and have it work the same with both C libraries? Might be nice to see how semihosting works in the sample though; it's a very useful tool in early SoC bring up as you don't need the clock tree set up correctly and driving a UART when working with real hardware.

Reply:
Noted. I will address this later on.